Use ansible_python_interpreter=auto

jsf9k opened the pull request:
## 🗣 Description

This pull request makes the change of using `ansible_python_interpreter=auto` wherever possible.  I also went ahead and pulled in the latest upstream changes from [cisagov/skeleton-packer](https://github.com/cisagov/skeleton-packer).

## 💭 Motivation and Context

In the future, `ansible_python_interpreter=auto` will be Ansible's default behavior, so it makes sense to go ahead and prepare for it.  Since we remove python2, it should not change the AMI that is generated.

## 🧪 Testing

I have successfully generated a staging AMI with these changes.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

mcdonnnj committed:
Ran `pre-commit autoupdate` to get the latest version of the
`pre-commit-terraform` hook since the PR I submitted,
https://github.com/antonbabenko/pre-commit-terraform/pull/100,
was approved. This will fix issues with `skeleton-tf-module` related to multiple
directories with Terraform code.

mcdonnnj committed:
Explicitly install pre-commit hooks as its own step so it's clearer when the
failure is with setup rather than hooks running.

mcdonnnj committed:
Rename action to accurately reflect that we are only linting, not building.
Adjust description for pre-commit hook setup to better convey what is being
done.

mcdonnnj committed:
Sort .gitignore entries.

mcdonnnj committed:
Merge pull request #41 from cisagov/bugfix/update_pre-commit_for_pre-commit-terraform_hook

Update pre-commit Hooks to Include Bugfix

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_upstream_updates

mcdonnnj committed:
Add Terraform related installation and initialization from skeleton-tf-module
to the 'lint' Action to provide necessary `pre-commit` elements.

mcdonnnj committed:
Rename the 'terraform' directory to 'terraform-test-user' to line up with the
'skeleton-packer-cool' directory structure.

mcdonnnj committed:
Add '.terraform' directories to the '.gitignore' file.

hillaryj committed:
Update CODEOWNERS

mcdonnnj committed:
Add the 'terraform-post-packer' direction from skeleton-packer-cool.

mcdonnnj committed:
Update the 'terraform-test-user' Terraform code to mirror skeleton-packer-cool
methods.

mcdonnnj committed:
Fix references to skeleton-packer-cool in the 'terraform-post-packer' directory.

mcdonnnj committed:
Update the prerelease workflow to mirror skeleton-packer-cool.

mcdonnnj committed:
Update the release workflow to mirror skeleton-packer-cool.

hillaryj committed:
Merge pull request #42 from cisagov/hillaryj-owner

Update CODEOWNERS

felddy committed:
Allow events from apb to rebuild this repository weekly.

felddy committed:
Merge pull request #43 from cisagov/improvement/apb-events

Allow events from apb to rebuild this repository weekly.

mcdonnnj committed:
Remove the terraform-post-packer directory because it relies on the COOL account
structure to function. The base `skeleton-packer` should still play nice with
non-COOL projects for the time being.
Update the `terraform-test-user` Terraform:
-Move the provider blocks from `main.tf` into their own file (`providers.tf`).
-Move the `caller_user_name` local definition into `providers.tf`.
-Move the `aws_caller_identity` block into its own file (`caller_identity.tf`).
-Fix the `caller_user_name` local variable to correctly indicate the username
 of the caller. Previously the split would return the name of the assumed role
 instead of the username of the caller.

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_upstream_updates

mcdonnnj committed:
Update theGitHub Actions workflow to use actions/setup-go@v2 instead of v2-beta.
Update the version of Go installed to the latest stable version.
Update the installation of terraform-docs so that it correctl installs.
Please see https://github.com/cisagov/skeleton-tf-module/pull/19 for more
information about this change.

mcdonnnj committed:
Added jsf9k's fix for Terraform code that uses `remote_state` not playing well
with the current `terraform validate` implementation:
I am commenting out this hook because terraform validate cannot currently run
without accessing the remote state because of the way the providers are defined
in providers.tf. This is something that may be remedied in the future. For more
information, check out these two GitHub issues:
* https://github.com/hashicorp/terraform/issues/15895
* https://github.com/hashicorp/terraform/issues/15811

mcdonnnj committed:
Re-add `terraform-post-packer` since the priority is COOL environment needs.
Update the README to correctly reflect building this project as it is now.
Add tweaks from the `packer.json` configuration in `skeleton-packer-cool`.

mcdonnnj committed:
Fix some dangling `skeleton-packer-cool` references.

mcdonnnj committed:
Adjust the line break for pre-commit in the GitHub Actions workflow.

mcdonnnj committed:
Remove line break character for pre-commit in GitHub Actions workflow.

mcdonnnj committed:
Rename `terraform.tf` files to `backend.tf` for consistency with other projects
per feedback.

mcdonnnj committed:
Break out providers and caller_identify from `main.tf` to their own files in the
`terraform-post-packer` directory.

mcdonnnj committed:
Modify the GitHub Actions workflow to generate a temporary pre-commit
configuration file with the terraform_validate hook removed.
Switch the GitHub Actions run of pre-commit to use the generated file instead.

mcdonnnj committed:
Correct the extension for the pre-commit configuration referenced in the
GitHub Actions workflow from .yml to .yaml

mcdonnnj committed:
Per discussion and feedback the terraform-test-user directory has been renamed
to terraform-build-user. Additionally all references to `test-skeleton-packer`
have been changed to `build-skeleton-packer`. A `terraform destroy` and
`terraform apply` have been run to correct these resources in AWS.

mcdonnnj committed:
Fix the terraform-build-user state key to reflect the transition from test-user
to build-user.

mcdonnnj committed:
Fix a dangling test-user reference in the README. With this I think all
references to test-user have appropriately been updated (famous last words).

jsf9k committed:
Use ansible_python_interpreter=auto in all playbooks

jsf9k committed:
Bump version from 0.2.0-rc.2 to 0.2.0-rc.3

mcdonnnj committed:
Merge pull request #34 from cisagov/improvements/pull_upstream_updates

Pull in Upstream Updates

dav3r reviewed (approved):
Defend today, secure tomorrow! 🦅

jsf9k committed:
Bump version from 0.2.0-rc.3 to 0.2.0

mcdonnnj reviewed (approved):
The times they are a-changin' 🎶

jsf9k committed:
Merge remote-tracking branch 'skeleton-packer/develop' into improvement/use-ansible-interpreter-auto

dav3r commented on terraform-post-packer/backend.tf:
```diff
@@ -5,6 +5,6 @@ terraform {
     profile        = "cool-terraform-backend"
     region         = "us-east-1"
-    key            = "freeipa-server-packer/terraform-test-user.tfstate"
+    key            = "skeleton-packer/terraform-post-packer.tfstate"
```
This key should be: `freeipa-server-packer/terraform-post-packer.tfstate`

jsf9k replied:
Please have a look at commit 8ca1ace and let me know if I caught everything.

dav3r replied:
Yes, that all looks correct to me- good catch on the providers and caller identity!

felddy reviewed (approved):
Looks good.  See @dav3r 's mod.

dav3r reviewed (approved):
👍 with one small, but important change requested.

jsf9k committed:
Correct key in backend state

Also remove providers and called identity from main.yml since those
things already exist in providers.tf and called-identity.tf,
respectively.

jsf9k committed:
Delete the correct ssh user

The admin user does not exist on Fedora.  The same user is called
fedora here.
